src/Uno.UI/UI/Xaml/Internal/EventManager.uno.cs

```diff
@@ -70,6 +70,12 @@ public void RaiseLayoutUpdated()
 		// This is actually what we want and what WinUI does.
 		foreach (var item in _layoutUpdatedSubscribers)
 		{
+			// Sometimes, we are racing with GC and the DependencyObjectStore is disposed (via finalizer)
+			if (((IDependencyObjectStoreProvider)item.Key).Store.IsDisposed)
```

Comment:
Yes, in that case, it's best to avoid failing with an exception if we know better.

Reply:
Looks like it's not enough 😕 

It just made the issue less frequent, but CI has two occurrences of the exception (even though no tests failed)

```
fail: Uno.UI.Dispatching.NativeDispatcher[0]
      NativeDispatcher unhandled exception
      System.InvalidOperationException: DependencyPropertyDetailsCollection is accessed after it was disposed.
         at Microsoft.UI.Xaml.DependencyPropertyDetailsCollection.TryGetPropertyDetails(DependencyProperty property, Boolean forceCreate) in C:\a\1\s\src\Uno.UI\UI\Xaml\DependencyPropertyDetailsCollection.cs:line 139
         at Microsoft.UI.Xaml.DependencyObjectStore.GetValue(DependencyProperty property) in C:\a\1\s\src\Uno.UI\UI\Xaml\DependencyObjectStore.cs:line 248
         at Microsoft.UI.Xaml.Controls.ContentPresenter.ArrangeNativeElement() in C:\a\1\s\src\Uno.UI\UI\Xaml\Controls\ContentPresenter\ContentPresenter.skia.cs:line 103
         at Uno.UI.Dispatching.NativeDispatcher.DispatchItems() in C:\a\1\s\src\Uno.UI.Dispatching\Native\NativeDispatcher.cs:line 134
fail: Uno.UI.Dispatching.NativeDispatcher[0]
      NativeDispatcher unhandled exception
      System.InvalidOperationException: DependencyPropertyDetailsCollection is accessed after it was disposed.
         at Microsoft.UI.Xaml.DependencyPropertyDetailsCollection.TryGetPropertyDetails(DependencyProperty property, Boolean forceCreate) in C:\a\1\s\src\Uno.UI\UI\Xaml\DependencyPropertyDetailsCollection.cs:line 139
         at Microsoft.UI.Xaml.DependencyObjectStore.GetValue(DependencyProperty property) in C:\a\1\s\src\Uno.UI\UI\Xaml\DependencyObjectStore.cs:line 248
         at Microsoft.UI.Xaml.Controls.ContentPresenter.ArrangeNativeElement() in C:\a\1\s\src\Uno.UI\UI\Xaml\Controls\ContentPresenter\ContentPresenter.skia.cs:line 103
         at Uno.UI.Dispatching.NativeDispatcher.DispatchItems() in C:\a\1\s\src\Uno.UI.Dispatching\Native\NativeDispatcher.cs:line 134
```

Will likely need to discuss this with you and/or @ebariche

Reply:
Thinking more about it, allowing a use-after-free may actually be fine as long as we just return without doing anything. What do you think?

Reply:
Agreed. I don't think we need to be failing in most cases. No-ops are better.